### PR TITLE
Add Instanovo

### PR DESCRIPTION
**Pull request by BioGeek**
Add [instanovo](https://github.com/instadeepai/InstaNovo), a package for _de novo_ peptide sequencing

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by BioGeek** [comment]
@coderabbitai full review

**Comment by hechth** [comment]
@hechth for reference

**Comment by hechth** [comment]
@BioGeek I've updated matchms' dependencies to newer versions (py>3.10, numpy>2, numba>0.60). Currently I'm waiting for sparsestack to get updated on conda-forge, then I will make the next release for matchms with the newer dependencies and then will update the matchms bioconda package. Then you can base your package on matchms >=0.28.0 and I hope things should be fine.

**Comment by BioGeek** [comment]
@hechth Thanks for the support!

File: recipes/instanovo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - python >=3.10,<3.12
+    - setuptools >=69.1.1
+    - pip
+  run:
+    - click >=8.1.7
+    - datasets >=3.0.1
+    - hydra-core >=1.3.2
+    - jaxtyping >=0.2.34
+    - jiwer >=3.0.4
+    - matchms >=0.27.0
```
Comment (hechth) [inline recipes/instanovo/meta.yaml:31]:
```suggestion
    - matchms >=0.28.1
```

File: recipes/instanovo/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - neptune >=1.12.0
+    - numpy >=1.23,<1.27
```
Comment (hechth) [inline recipes/instanovo/meta.yaml:33]:
```suggestion
    - numpy >=2.0.0
```

**Comment by hechth** [comment]
@BioGeek seems like pyOpenMS is also on numpy < 2 - do you really need that library?

**Comment by hechth** [comment]
xref https://github.com/bioconda/bioconda-recipes/pull/51992

**Comment by hechth** [comment]
@BioGeek seems like the problem is now somewhere with cuda - maybe @bgruening can confirm or has an idea of how to fix this? I did not wrap any packages using CUDA before.

**Comment by BioGeek** [comment]
@hechth , I [tried](https://github.com/bioconda/bioconda-recipes/pull/51796/commits/01c40d1b6758b4c5585b0c96b8356b83bd7a0dd6) with using `pytorch` instead of `pytorch-gpu` , but that doesn't seem to have helped either.

Thanks for your support so far already!

**Comment by mfansler** [comment]
This seems all driven by `numpy >=2` specification. Is that really necessary?

Could you point out the requirements bounds from upstream that you are working from? Running `grayskull` to generate a skeleton leaves everything unbounded. I'd recommend dropping all bounds and seeing what happens.

**Comment by BioGeek** [comment]
@mfansler Thanks for stopping by and taking a look!

[Removing all the bounds](https://github.com/bioconda/bioconda-recipes/pull/51796/commits/2587292cfe10a96ce5e61aee2ca4571f0d7820e2) seems to have found a combination that works with `numpy 1.23.5`, but then [fails](https://github.com/bioconda/bioconda-recipes/actions/runs/11872810170/job/33086971804?pr=51796) at the `pip check` stage with:

```
20:04:35 BIOCONDA INFO (OUT) matchms 0.27.0 requires rdkit, which is not installed.
20:04:35 BIOCONDA INFO (OUT) matchms 0.27.0 has requirement lxml<5.0.0,>=4.9.3, but you have lxml 5.2.2.
20:04:35 BIOCONDA INFO (OUT) matchms 0.27.0 has requirement scipy>=1.11, but you have scipy 1.10.1.
20:04:35 BIOCONDA INFO (OUT) PubChemPy 1.0.4 is not supported on this platform
```

**Comment by BioGeek** [comment]
@BiocondaBot please add label

**Review by mfansler — APPROVED** [review]
LGTM! Thanks for taking time to submit this. ❤️

File: recipes/instanovo/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    - pyopenms
+    - python-dotenv
+    - pytorch-lightning
+    - s3fs
+    - scikit-learn
+    - spectrum_utils
+    - tensorboard
+    - pytorch
+    - tqdm
+    - pytorch-gpu
```
[the remark this answers is not in the document]
Comment (BioGeek) [reply recipes/instanovo/meta.yaml]: So specifying only `pytorch` [didn't seem to have worked](https://github.com/bioconda/bioconda-recipes/pull/52231).  The cpu version of `pytorch` is installed even though CUDA drivers and an NVIDIA GPU are availble on the system.

**Pull request merged**